### PR TITLE
#5333 fixed wrong redirect to home page on map logout

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
The redirect is probably due to a cut and paste from the same work of Home button. 
To test it I had to increase the PluginTestUtils to include also reducers coming from additional plugins. 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#5333

**What is the new behavior?**
Now logout from map, when confirm to not save changes, does not redirect to the home page.
## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/plugins/__tests__/Login-test.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            expect(document.querySelector('#mapstore-navbar-container .glyphicon-user')).toBeTruthy();
+            // show modal
+            expect(document.querySelector('.ms-resizable-modal')).toExist();
+            const buttons = document.querySelectorAll('.ms-resizable-modal .modal-footer button');
+            expect(buttons[0]).toBeTruthy();
+            expect(buttons[1]).toBeTruthy();
+            // click on confirm button
+            TestUtils.Simulate.click(buttons[0]);
+        });
+        it('confirm dialog closes dialog', done => {
```
Comment (offtherailz) [inline web/client/plugins/__tests__/Login-test.js:101]:
```suggestion
        it('not confirm closes dialog', done => {
```

File: web/client/plugins/OmniBar.jsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ class OmniBar extends React.Component {
                 current.tools.map((tool, index) => ({
                     name: current.name + index,
                     panel: tool,
-                    cfg: current.cfg.toolsCfg ? current.cfg.toolsCfg[index] : {}
+                    cfg: current?.cfg?.toolsCfg ? current.cfg.toolsCfg[index] : {}
```
Comment (MV88) [inline web/client/plugins/OmniBar.jsx:38]: What about this?
```suggestion
                    cfg: current?.cfg?.toolsCfg[index] || {}
```

File: web/client/plugins/__tests__/Login-test.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            const { Plugin: OmniBarPlugin } = getPluginForTest(OmniBar, storeState, { LoginPlugin: Login });
+            ReactDOM.render(<OmniBarPlugin items={[{ ...Login.LoginPlugin.OmniBar, plugin: Plugin.LoginPlugin}]} />, document.getElementById("container"));
+            expect(document.querySelector('#mapstore-navbar-container .glyphicon-user')).toBeTruthy();
+        });
+        it('confirm dialog on pending changes', done => {
+            const storeState = stateMocker(
+                loginSuccess({  User: { name: "Test", access_token: "some-token"}}),
+                setControlProperty('unsavedMap', 'enabled', true),
+                setControlProperty('unsavedMap', 'source', 'logout')
+                // TODO: not sure about why this is required, investigate.
```
Comment (MV88) [inline web/client/plugins/__tests__/Login-test.js:80]: what about this todo?

File: web/client/plugins/login/index.js [hunk 4]
```diff
@@ -76,7 +75,7 @@ const LoginNav = connect((state) => ({
     renderUnsavedMapChangesDialog: ConfigUtils.getConfigProp('unsavedMapChangesDialog'),
     displayUnsavedDialog: unsavedMapSelector(state)
         && unsavedMapSourceSelector(state) === 'logout'
-        && feedbackMaskSelector(state).currentPage === 'viewer'
+        // && pathnameSelector(state).indexOf("viewer") >= 0
```
Comment (MV88) [inline web/client/plugins/login/index.js:78]: please remove this if it is not needed

File: web/client/plugins/__tests__/Login-test.js (hunk 3 above)
Comment (offtherailz) [reply web/client/plugins/__tests__/Login-test.js:80]: Good catch. I removed that dependency at the end, but not the TODO :smile: 
```suggestion
```

**Comment by MV88** [comment]
please provide a backport 2020.02.xx

**Pull request merged**